geoapps/io/validators.py

```diff
@@ -249,13 +249,13 @@ def _validate_parameter_uuid(
                 raise IndexError(msg)
 
     def _general_validation_msg(self, param: str, type: str, value: Any) -> str:
-        """ Generate base error message: "Invalid '{param}' {type}: {value}.". """
+        """Generate base error message: "Invalid '{param}' {type}: {value}."."""
         return f"Invalid '{param}' {type}: '{value}'."
 
     def _iterable_validation_msg(
         self, param: str, type: str, value: Any, validations: List[Any]
```

Comment:
Why is `List` accepted here and not `list`?

Reply:
mmh, difference between `typing.List` and `list` is that `typing.List` is generic and allows to specify item type.
But since Python 3.9, we can do the same with `list`.
So I have discovered that pre-commit was running on python 3.9. I will force it to 3.7 and see what it gives out...

Reply:
@domfournier , simpler type hint are available only if doing `from __future__ import annotations`
That was missing from this file. Fixed in https://github.com/MiraGeoscience/geoapps/pull/124
Especially see the simple syntax for `Union`